cluster-autoscaler/utils/taints/taints.go

```diff
@@ -215,22 +211,34 @@ func AddTaint(node *apiv1.Node, client kube_client.Interface, taint apiv1.Taint,
 		}
 
 		if err != nil {
-			klog.Warningf("Error while adding %v taint on node %v: %v", getKeyShortName(taint.Key), node.Name, err)
+			klog.Warningf("Error while adding %v taints on node %v: %v", strings.Join(taintKeys(taints), ","), node.Name, err)
 			return err
 		}
-		klog.V(1).Infof("Successfully added %v on node %v", getKeyShortName(taint.Key), node.Name)
+		klog.V(1).Infof("Successfully added %v on node %v", strings.Join(taintKeys(taints), ","), node.Name)
 		return nil
 	}
 }
 
-func addTaintToSpec(node *apiv1.Node, taint apiv1.Taint, cordonNode bool) bool {
-	for _, t := range node.Spec.Taints {
-		if t.Key == taint.Key {
-			klog.V(2).Infof("%v already present on node %v, t: %v", taint.Key, node.Name, t)
-			return false
+func addTaintsToSpec(node *apiv1.Node, taints []apiv1.Taint, cordonNode bool) bool {
+	taintsAdded := false
+	for _, taint := range taints {
+		taintPresent := false
```

Comment:
Instead of a nested loop, it would be a bit more readable to use `HasTaint` function which already exists in this file.